Consolidate BREAKING.md files

scottn12 opened the pull request:
## Description

This PR consolidates BREAKING.md files for non-server packages. It also updates the guidelines for a breaking change note outlined in BREAKING.md.

In addition, a new wiki page was created [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes). If possible, please take some time to read through it and let me know if you have any feedback.

## Reviewer Guidance

- Any other guidelines we should add?
- Should we update the headers for each section? I think they can be a little confusing for newer folks, but I don't have a better idea as of now. For example, for reference this is the current format:
```markdown
# 2.0.0
## 2.0.0 Upcoming Changes
...
## 2.0.0 Breaking Changes
...

# 1.2.0
## 1.2.0 Upcoming Changes
...

# 1.1.0
## 1.1.0 Upcoming Changes
...

# 1.0.0
## 1.0.0 Upcoming Changes
...
## 1.0.0 Breaking Changes
...
``` 



## Other information or known dependencies

[AB#1034](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1034)
[AB#1069](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1069)

scottn12 committed:
update instructions for notes, remove extra breaking.md files

scottn12 committed:
add link to what's new on FF.com

ssimic2 reviewed (approved):
We chatted about pulling common-defs back (its separate release group). The rest is good.

scottn12 committed:
revert common-definitions breaking.md

revert common-definitions breaking.md

revert common-definitions breaking.md

revert common-definitions breaking.md

scottn12 committed:
revert common-definitions breaking.md

scottn12 committed:
revert common-definitions breaking.md

scottn12 committed:
revert common-definitions breaking.md

scottn12 committed:
revert common-definitions breaking.md

scottn12 committed:
Merge branch 'breakingMDChanges' of https://github.com/scottn12/FluidFramework into breakingMDChanges

scottn12 committed:
revert common-definitions breaking.md
